src/routes/_utils/scrollIntoView.js

```diff
@@ -0,0 +1,64 @@
+import {
+  getScrollContainer,
+  getOffsetHeight } from './scrollContainer'
+import { smoothScroll } from './smoothScroll'
+
+function getTopOverlay () {
+  return document.getElementById('main-nav').clientHeight
+}
+
+export function isVisible (element) {
+  if (!element) {
+    return false
+  }
+  let rect = element.getBoundingClientRect()
+  let offsetHeight = getOffsetHeight()
+  let topOverlay = getTopOverlay()
+  return rect.top < offsetHeight && rect.bottom >= topOverlay
+}
+
+export function firstVisibleElementIndex (items, itemElementFunction) {
```

Comment:
Wondering if we could do a binary search here instead of checking each element one at a time, but that may be an excessive micro-optimization.

Reply:
This is possible. I'll give it a try to see how much it improves things. Whether it'll be worth it depends on how big the array of visible items can grow on virtual lists.

Reply:
I tried to address this in #910. Let's continue the conversation there.